### DISCUSSION TITLE
Question: From Compose Multiplatform iOS Android Template to Decompose

### DISCUSSION THREAD

**Discussion by TimOrtel** [arkivanov/Decompose#377] [Q&A]
Using this template https://github.com/JetBrains/compose-multiplatform-ios-android-template I created a project, however, it is not clear to me how I can create my RootComponent in the iOS target and pass it to my compose code. Maybe the documentation on this could be improved.

**Comment by arkivanov** [comment]
Yeah, it's pretty difficult to cover all possible patterns and templates. I think you should be able to create `RootComponent` in [main.ios.kt](https://github.com/JetBrains/compose-multiplatform-ios-android-template/blob/main/shared/src/iosMain/kotlin/main.ios.kt).

Something like this should work (from the top of my head):

```kotlin
fun MainViewController(): UIViewController {
    val lifecycle = LifecycleRegistry()
    val context = DefaultComponentContext(lifecycle = lifecycle)
    val root = DefaultRootComponent(context = context)

    return ComposeUIViewController {
        App(root = root)

        DisposableEffect {
            lifecycle.resume()
            onDispose { lifecycle.destroy() }
        } 
    }
}
```

Also, there could be away to bind `LifecycleRegistry` with the returned `UIViewController` so that it calls `resume` and `stop` when the app goes to foreground and background. But I'm not sure how to it yet.
